splice_machine/src/test/java/com/splicemachine/derby/impl/sql/actions/index/IndexIT.java

```diff
@@ -964,6 +964,21 @@ public void testCreateIndexAndInsertWithExpressionsNullInput() throws Exception
          */
     }
 
+    // DB-10431
+    @Test
+    public void testCreateIndexAndInsertWithExpressionsBatchWithFullNullRow() throws Exception {
+        String tableName = "TEST_IDX_BATCH_INSERT";
+        methodWatcher.executeUpdate(format("create table %s (c char(4), i int, d double)", tableName));
+        methodWatcher.executeUpdate(format("insert into %s values ('abc', 10, 1.1), ('def', 20, 2.2)", tableName));
+
+        methodWatcher.executeUpdate(format("CREATE INDEX %s_IDX ON %s (upper(c), i + 2)", tableName, tableName));
+
+        // Without the fix, the following insert fails sporadically. It depends on the order of inserting rows. If
+        // (NULL, NULL, NULL) is inserted first, it fails. With the fix, it should never fail.
+        methodWatcher.executeUpdate(format("insert into %s values ('jkl', 30, 2.2), (NULL, NULL, NULL)", tableName));
+        methodWatcher.executeUpdate(format("update %s set c = 'def' where i = 10", tableName));
```

Comment:
could you also verify the results of these operations?

Reply:
Yes. Since we have only the index expr DDL/DML changes in master, we cannot query using the index yet. Nevertheless, I verify the result by issuing a select without the expr index to make sure insert and update ops modify data as expected.